pkg/dockerutils/utils.go

```diff
@@ -1,29 +1,47 @@
 package dockerutils
 
 import (
+	"errors"
+	"io"
 	"io/ioutil"
 	"os"
 	"regexp"
 )
 
+var ErrContainerIDNotFound = errors.New("container id not found")
+
 // CurrentContainerID returns the current running container ID
 func CurrentContainerID() (string, error) {
-	f, err := os.Open("/proc/self/cgroup")
+	fCgroup, err := os.Open("/proc/self/cgroup")
+	if err != nil {
+		return "", err
+	}
+
+	fMountInfo, err := os.Open("/proc/self/mountinfo")
 	if err != nil {
 		return "", err
 	}
-	defer f.Close()
 
-	content, err := ioutil.ReadAll(f)
+	defer fCgroup.Close()
+	defer fMountInfo.Close()
+
+	reader := io.MultiReader(fCgroup, fMountInfo)
+
+	content, err := ioutil.ReadAll(reader)
 	if err != nil {
 		return "", err
 	}
 
-	re := regexp.MustCompilePOSIX(`[0-9]+:[a-z_,=]+.*docker[/-]([0-9a-f]{64})`)
+	re := regexp.MustCompilePOSIX(`([0-9]+:[a-z_,=]+.*docker[/-]|/var/lib/docker/containers)?([0-9a-f]{64})`)
 	line := re.FindAllSubmatch(content, -1)
-	if len(line) <= 0 || len(line[0]) != 2 {
-		return "", nil
+	if len(line) <= 0 || len(line[0]) != 3 {
```

Comment:
This is confusing. Why 3? Why <= 0.

Reply:
I'm using ```len(line[0]) != 3 ``` because now the regexp has 2 grouping 
```	re := regexp.MustCompilePOSIX(`([0-9]+:[a-z_,=]+.*docker[/-]|/var/lib/docker/containers/)([0-9a-f]{64})`)```
in golang regexp documentation sintax there's a  non-capturing group syntax but does not work here

about len(line) <= 0  I will see it's necessary

Reply:
if matches == nil || len(matches) != 3 {
   return NotFound
}